fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1949,8 +2077,58 @@ public Builder forbidRecordScan() {
                 return this;
             }
 
+
+            /**
+             * Set the desired action if the index is disabled.
+             * @param ifDisabled CONTINUE: build the index (effectively, it's the same as rebuild)
+             *                   REBUILD: rebuild the index.
+             *                   ERROR: reject (with exception).
+             * @return this builder
+             */
+            public Builder setIfDisabled(final DesiredAction ifDisabled) {
+                this.ifDisabled = ifDisabled;
+                return this;
+            }
+
+            /**
+             * Set the desired action if the index is in a write-only mode (i.e. partly built).
+             * @param ifWriteOnly CONTINUE: (default) attempt to continue the previous indexing process
+             *                    REBUILD: clear the existing data and rebuild
+             *                    ERROR: reject (with exception).
+             * @return this builder
+             */
+            public Builder setIfWriteOnly(final DesiredAction ifWriteOnly) {
+                this.ifWriteOnly = ifWriteOnly;
+                return this;
+            }
+
+            /**
+             * Set the desired action if the index is in a write-only mode but the previous indexing method mismatches the requested one.
+             * @param ifMismatchPrev CONTINUE: (default) attempt to continue the previous indexing process, using the previous policy
+             *                       REBUILD: clear the existing data and rebuild, using the new policy
+             *                       ERROR: reject (with exception).
+             * @return this builder
+             */
+            public Builder setIfMismatchPrev(final DesiredAction ifMismatchPrev) {
```

Comment:
Sorry, I should have mentioned this earlier, but maybe this should be `setIfMismatchedPrevious` or maybe even `setIfPolicyChanged`. It might also be fine as is; I don't have super strong feelings on this.

Reply:
Sure, NP. Changed it to the your first suggestion.